apps/src/netsim/NetSimMessage.js

```diff
@@ -104,7 +104,20 @@ NetSimMessage.send = function (shard, messageData, onComplete) {
   entity.payload = messageData.payload;
   entity.extraHopsRemaining = utils.valueOr(messageData.extraHopsRemaining, 0);
   entity.visitedNodeIDs = utils.valueOr(messageData.visitedNodeIDs, []);
-  entity.getTable().create(entity.buildRow(), onComplete);
+  try {
+    entity.getTable().create(entity.buildRow(), onComplete);
+  } catch (err) {
+    onComplete(err, null);
+  }
+};
+
+/**
+ * Static helper.
+ * @param {NetSimMessage} message
+ * @returns {boolean} TRUE iff the given message is well-formed.
+ */
+NetSimMessage.isValid = function (message) {
+  return /^[01]*$/.test(message.payload);
```

Comment:
0 length is valid. We could make it invalid if we thought that was a better user experience, but I think doing so is pretty low priority.